xesmf/cf.py

```diff
@@ -0,0 +1,203 @@
+"""
+Decode longitude and latitude of an `xarray.Dataset` or a `xarray.DataArray`
+according to CF conventions.
+"""
+
+import warnings
+
+CF_SPECS = {
+    'lon': {
+        'name': ['lon', 'longitude'],
+        'units': ['degrees_east', 'degree_east', 'degree_E', 'degrees_E',
+                  'degreeE', 'degreesE'],
+        'standard_name': ['longitude'],
+        'axis': 'X'
+        },
+    'lat': {
+        'name': ['lat', 'latitude'],
+        'units': ['degrees_north', 'degree_north', 'degree_N', 'degrees_N',
+                  'degreeN', 'degreesN'],
+        'standard_name': ['latitude'],
+        'axis': 'Y'
+        }
+    }
+
+
+def get_coord_name_from_specs(ds, specs):
```

Comment:
This is certainly a more extensive search than what #74 provides. The question is do we want this ? Also, from a user perspective, all the different ways coordinates can be identified might create some confusion.

Reply:
In fact, it is the strict application of the CF conventions, which must be considered to be the reference for formatting dataset, and which are increasingly being followed.

Reply:
See https://github.com/JiaweiZhuang/xESMF/pull/94#issuecomment-618699019

Reply:
Yep. Checking with units first since they are mandatory. And then standard_name, just in case.